device-telemetry/app/com/microsoft/azure/iotsolutions/devicetelemetry/services/runtime/MessagesConfig.java

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+package com.microsoft.azure.iotsolutions.devicetelemetry.services.runtime;
+
+import play.Logger;
+
+public class MessagesConfig {
+
+    private static final Logger.ALogger log = Logger.of(MessagesConfig.class);
+
+    private final String storageType;
+    private final StorageConfig storageConfig;
+    private final TimeSeriesConfig timeSeriesConfig;
+
+    public MessagesConfig(
+        String storageType,
+        StorageConfig storageConfig,
+        TimeSeriesConfig timeSeriesConfig) {
+
+        this.storageType = storageType;
+        this.storageConfig = storageConfig;
+        this.timeSeriesConfig = timeSeriesConfig;
+    }
+
+    public StorageType getStorageType() {
+        switch (this.storageType.toLowerCase()) {
+            case "tsi":
+            case "timeseries":
+            case "timeseriesinsights":
+                return StorageType.tsi;
+            default:
+               return StorageType.cosmosdb;
+        }
+    }
+
+    public StorageConfig getStorageConfig() {
```

Comment:
The two config are different set of parameters. Dictionary only hide the difference 
 complexity in getStorageConfig(). But it will still need to know the details when using these different values.